### ISSUE TITLE
Error executing 'postInstallation': Server returned status code 500 from http://127.0.0.1/oc-includes/osclass/install.php

### ISSUE THREAD

**Issue by martin-kudlicka** [bitnami/bitnami-docker-osclass#35]
**Description**

Hi. This is probably similar to issue #32 . I am trying to use bitnami/osclass:3.7.4-r188 image and I am getting this output when started:

```
osclasstest   |
osclasstest   | Welcome to the Bitnami osclass container
osclasstest   | Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-osclass
osclasstest   | Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-osclass/issues
osclasstest   |
osclasstest   | nami    INFO  Initializing apache
osclasstest   | apache  INFO  ==> Patching httpoxy...
osclasstest   | apache  INFO  ==> Configuring dummy certificates...
osclasstest   | nami    INFO  apache successfully initialized
osclasstest   | nami    INFO  Initializing php
osclasstest   | nami    INFO  php successfully initialized
osclasstest   | nami    INFO  Initializing mysql-client
osclasstest   | mysql-c INFO  Trying to connect to MySQL server
osclasstest   | mysql-c INFO  Found MySQL server listening at mariadb:3306
osclasstest   | mysql-c INFO  MySQL server listening and working at mariadb:3306
osclasstest   | mysql-c INFO  ==> Found database osclasstest_cz. Skipping database creation...
osclasstest   | nami    INFO  mysql-client successfully initialized
osclasstest   | nami    INFO  Initializing libphp
osclasstest   | nami    INFO  libphp successfully initialized
osclasstest   | nami    INFO  Initializing osclass
osclasstest   | osclass INFO  Creating the database...
osclasstest   | mysql-c INFO  Trying to connect to MySQL server
osclasstest   | mysql-c INFO  Found MySQL server listening at mariadb:3306
osclasstest   | mysql-c INFO  MySQL server listening and working at mariadb:3306
osclasstest   | osclass INFO  Preparing webserver environment...
osclasstest   | osclass INFO  Installing Osclass...
osclasstest   | Error executing 'postInstallation': Server returned status code 500 from http://127.0.0.1/oc-includes/osclass/install.php
osclasstest   | Server returned status code 500 from http://127.0.0.1/oc-includes/osclass/install.php
osclasstest exited with code 1
```

**Steps to reproduce the issue:**

1. `docker-compose -f osclasstest.yml up`

**Describe the results you received:**

Installation is not successful and osclass is not working. When I start the container again without clearing database (which was initialized in first run) I get another error, same as in issue #32 

```
osclasstest   |
osclasstest   | Welcome to the Bitnami osclass container
osclasstest   | Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-osclass
osclasstest   | Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-osclass/issues
osclasstest   |
osclasstest   | nami    INFO  Initializing osclass
osclasstest   | osclass INFO  Creating the database...
osclasstest   | mysql-c INFO  Trying to connect to MySQL server
osclasstest   | mysql-c INFO  Found MySQL server listening at mariadb:3306
osclasstest   | mysql-c INFO  MySQL server listening and working at mariadb:3306
osclasstest   | osclass INFO  Preparing webserver environment...
osclasstest   | osclass INFO  Installing Osclass...
osclasstest   | Error executing 'postInstallation': No INPUT matching 's_name'
osclasstest exited with code 1
```

**Describe the results you expected:**

Osclass successfully installed and working.

**Additional information you deem important (e.g. issue happens only occasionally):**

Issue happens always. I tried several other revisions of version 3.7.4 (1, 50, 150), all with the same result.

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        6247962
 Built:             Sun Feb 10 04:12:31 2019
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:13:06 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

- Output of `docker info`:

```
Containers: 13
 Running: 7
 Paused: 0
 Stopped: 6
Images: 13
Server Version: 18.09.2
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9754871865f7fe2f4e74d43e2fc7ccd237edcbce
runc version: 09c8266bf2fcf9519a651b04ae54c967b9ab86ec
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.125-linuxkit
Operating System: Docker for Windows
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 3.161GiB
Name: linuxkit-00155dad162c
ID: SWNS:4SF6:RQNW:5UOS:2UNX:7R4I:IVII:VAPI:SGZF:GGZ3:KZAU:JK4X
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 92
 Goroutines: 147
 System Time: 2019-03-06T17:20:08.5860971Z
 EventsListeners: 8
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.23.2, build 1110ad01
docker-py version: 3.6.0
CPython version: 3.6.6
OpenSSL version: OpenSSL 1.0.2o  27 Mar 2018
```

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**

**Comment by martin-kudlicka** [comment]
I tried few more things and my issue is probably not connected with **bitnami** Osclass image but Osclass in Docker generally.

I was able to install successfully Osclass on Windows in Wampserver but when I export database from there to my Docker MariaDB instance it's not working anymore. Osclass gives me a message `Osclass isn't installed.` (I also copied and corrected config.php).

Then I tried **1and1internet** Docker nginx images and I get HTTP 500 error after first installation step (tables are created in database and next screen should be setup of web title, email etc.). It's probably same problem as I wrote in this issue - `Error executing 'postInstallation': Server returned status code 500...`.

It almost looks like Osclass is able to write to database but can't read from it. I checked in sources that `Osclass isn't installed.` happens when `osclass_installed` is not set or set to **0** in database but I can see it's set to **1**.

**Comment by martin-kudlicka** [comment]
from error.log
```
2019/03/06 20:35:42 [error] 44#44: *28 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Call to a member function row() on boolean in /var/www/html/oc-includes/osclass/model/Page.php:394
Stack trace:
#0 /var/www/html/oc-includes/osclass/install-functions.php(516): Page->insert(Array, Array)
#1 /var/www/html/oc-includes/osclass/install-functions.php(426): oc_install_example_data()
#2 /var/www/html/oc-includes/osclass/install.php(114): oc_install()
#3 {main}
  thrown in /var/www/html/oc-includes/osclass/model/Page.php on line 394" while reading response header from upstream, client: 172.18.0.1, server: _, request: "POST /oc-includes/osclass/install.php HTTP/1.1", upstream: "fastcgi://unix:/var/run/php/php7.2-fpm.sock:", host: "localhost", referrer: "http://localhost/oc-includes/osclass/install.php"
```

**Comment by tompizmor** [comment]
Hi @martin-kudlicka,

I just tried the docker-compose file from the [3/debian-9 folder](https://github.com/bitnami/bitnami-docker-osclass/blob/master/3/debian-9/docker-compose.yml) in this repo and it worked fine. The only modification I did was updating the `OSCLASS_HOST` environment variable.

Is your `osclasstest.yml` file different to the docker-compose file in the repo? if so, I can try to deploy it to see if I can reproduce the error.

**Comment by martin-kudlicka** [comment]
Hi,

thanks for reply. I tried that compose file with this result at the end
```
osclass_1   | mysql-c INFO  Trying to connect to MySQL server
osclass_1   | Error executing 'postInstallation': Failed to connect to mariadb:3306 after 36 tries
docker_osclass_1 exited with code 1
```
Seems like some internal network problem inside Docker container with database. But then I don't understand why my two Drupal instances works with MariaDB instance.

Just for curiosity I tried **mattrayner/lamp** with manually installed Osclass and it worked, maybe because everything including database is running within one container. But this solution is consuming too much memory.

Restarting Docker service didn't help... I will reboot PC and see what happens…

By the way, what did you update to OSCLASS_HOST variable? I tried just localhost and 127.0.0.1.

**Comment by martin-kudlicka** [comment]
bitnami-osclass.yml is working now, I forgot to change MARIADB_HOST there, I changed it to have different service name than my exsting mariadb (probably not needed).

**Comment by martin-kudlicka** [comment]
When I make one change in bitnami-osclass.yml - set mariadb volume instead of `mariadb_data:/bitnami` to external storage at `d:/Docker/mariadb/bitnami:/bitnami` it stops working again
```
osclass_1   | Error executing 'postInstallation': MySQL command failed to run. Error:
osclass_1   |  ERROR 1146 (42S02) at line 1: Table 'bitnami_osclass.oc_t_preference' doesn't exist
```
Tables are not created at all (bitnami_osclass folder is empty). Could this be the problem? I mean that volumes have to be **local** ?

**Comment by tompizmor** [comment]
> By the way, what did you update to OSCLASS_HOST variable? I tried just localhost and 127.0.0.1.

Because I was not using docker on my machine but in a remote server so in order to access I needed the external ip of the server. If you are running docker on your machine you are good using localhost or 127.0.0.1

>  Could this be the problem? I mean that volumes have to be local ?

What do you mean by local? Docker volumes or mount local directories from the host server? Both options worked for me.
For you external storage path (`d:/Docker/mariadb/bitnami:/bitnami`) I would say you are using Docker for Windows no?

**Comment by martin-kudlicka** [comment]
Hi again and thanks for patience and help. It's finally working, not the way I wanted, but I think problem is in database version.

In your osclasstest.yml you are using `bitnami/mariadb:latest` which seems to be version **10.1**. I am using official `mariadb:latest` which is version **10.4** and here I am getting those HTTP 500 in postInstallation step.

Then I tried last official `mysql:latest`. Here installation failed when trying to create database and user set in environment
```
osclass_1    | Error executing 'postInstallation': MySQL command failed to run. Error:
osclass_1    |  ERROR 1064 (42000) at line 1: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'IDENTIFIED BY 'osclasspassword'' at line 1
```
maybe also too new database version for osclass, anyway when I manually created user and granted him rights to osclass database, installation completed "succesfully" - there was empty (no categories) main page, categories also can't be created in administration (no error), so who knows what else is not working besides that.

`mariadb:10.3.13` seems to be working fine, main page contains categories and also I can create new one in settings.
